types/wlr_output.c

```diff
@@ -454,6 +453,10 @@ void wlr_output_cursor_set_surface(struct wlr_output_cursor *cursor,
 		wl_signal_add(&surface->events.destroy, &cursor->surface_destroy);
 		output_cursor_commit(cursor);
```

Comment:
Probably should check if you set the cursor with a null buffer and handle that situation.

Reply:
That's handled in `output_cursor_commit`.